### PR TITLE
Fix authentication failure loop

### PR DESCRIPTION
**Pull request by bryce-seifert**
Fixes https://github.com/bitfocus/companion-module-obs-studio/issues/60

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -101,9 +101,13 @@ instance.prototype.init = function() {
 			}
 		});
 
-		self.obs.on('SceneCollectionChanged', function() {
```
Comment (krocheck) [inline index.js:104]: Was the `SceneCollectionChanged` trigger not needed?

Comment (bryce-seifert) [reply index.js:104]: I thought I had just moved it down, so that the authentication action was organized with the connection actions.

Comment (bryce-seifert) [reply index.js:104]: Thanks for catching! Should be back now

Comment (krocheck) [reply index.js:104]: Previously, there were two of these, the 2nd one being further down and consisting of:
```
		self.obs.on('SceneCollectionChanged', (data) => {
			self.updateCurrentSceneCollection()
		})
```

That's a different call then this one, but I'm not sure if these are effectively duplicated in this branch?

Comment (bryce-seifert) [reply index.js:104]: It looks like there are 2 on the master branch as well, but I'm not entirely sure why. I can look into it. Looks like the second one got added here: https://github.com/bitfocus/companion-module-obs-studio/pull/46/commits/ff8d6d53a63640bcca19a547d512007e9857aec0

Comment (krocheck) [reply index.js:104]: Yeah, checking for duplicates like that is not something Jeffrey or I would typically scan for.  If you can clean that up/consolidate that would be helpful.  thx!

Comment (bryce-seifert) [reply index.js:104]: Ok while looking into the duplicate, it looks like  https://github.com/bitfocus/companion-module-obs-studio/commit/ff8d6d53a63640bcca19a547d512007e9857aec0 actually completely broke variables and I'm not sure why.

Comment (bryce-seifert) [reply index.js:104]: @krocheck Ok thanks for the pointers, it looks like variables (and other things) were being init everywhere but...at init. So I made that change below. Also tested consolidating the duplicated and that seems to work too.

Let me know if anything looks funky here.

**Pull request merged**